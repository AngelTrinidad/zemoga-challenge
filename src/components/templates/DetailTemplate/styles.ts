import {StyleSheet} from 'react-native';

import {DEFAULT_SPACES} from '~constants/styles';
import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  content: {
    padding: DEFAULT_SPACES.paddingScreen,
    paddingBottom: scaleSpace(24),
  },
});

export default styles;
