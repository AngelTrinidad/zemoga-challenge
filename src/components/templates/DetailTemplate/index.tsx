import React, {FC, PropsWithChildren} from 'react';
import {ScrollView, ViewStyle} from 'react-native';

import {useSafeAreaInsets} from 'react-native-safe-area-context';

import styles from './styles';

import {ViewFlex} from '~components/atoms';
import HeaderTitle from '~components/molecules/HeaderTitle';
import {useTheme} from '~contexts';
import {scaleSpace} from '~helpers';

interface Props extends PropsWithChildren {
  title?: string;
}

const DetailTemplate: FC<Props> = ({children, title = '', ...props}) => {
  const {top, bottom} = useSafeAreaInsets();
  const {colors} = useTheme();

  const headerStyles: ViewStyle = {
    paddingTop: top + scaleSpace(24),
  };

  const containerStyles: ViewStyle = {
    paddingBottom: bottom,
    backgroundColor: colors.screen.primary,
  };

  const contentStyles: ViewStyle = {
    backgroundColor: colors.screen.primary,
  };

  return (
    <ViewFlex style={containerStyles} {...props}>
      <HeaderTitle title={title} style={headerStyles} />
      <ScrollView
        contentContainerStyle={[contentStyles, styles.content]}
        showsVerticalScrollIndicator={false}>
        {children}
      </ScrollView>
    </ViewFlex>
  );
};

export default DetailTemplate;
