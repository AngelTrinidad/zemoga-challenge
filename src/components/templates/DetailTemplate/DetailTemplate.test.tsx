import React from 'react';
import {Text} from 'react-native';

import DetailTemplate from '.';

import {render} from '~helpers/testing';

describe('<DetailTemplate />', () => {
  let component: any;

  beforeEach(() => {
    component = render(
      <DetailTemplate title="Hello">
        <Text>Hello</Text>
      </DetailTemplate>,
    );
  });

  it('should render correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
