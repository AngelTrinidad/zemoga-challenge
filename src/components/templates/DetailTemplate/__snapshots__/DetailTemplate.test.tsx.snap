// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailTemplate /> should has same snapshot 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
      {
        "backgroundColor": "#F8F8F8",
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "paddingHorizontal": 22.857142857142858,
          "paddingVertical": 27.42857142857143,
        },
        {
          "backgroundColor": "#CFB997",
        },
        {
          "paddingTop": 27.42857142857143,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontFamily": "System",
          "fontSize": 29.485714285714288,
          "fontWeight": "800",
          "textAlign": undefined,
        }
      }
    >
      Hello
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "backgroundColor": "#F8F8F8",
        },
        {
          "padding": 22.857142857142858,
          "paddingBottom": 27.42857142857143,
        },
      ]
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <Text>
        Hello
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
